src/providers/42.ts

```diff
@@ -0,0 +1,71 @@
+import { OAuth2Client } from "oslo/oauth2";
+
+import type { OAuth2Provider } from "../index.js";
+
+export class FortyTwo implements OAuth2Provider {
+	private client: OAuth2Client;
+	private clientSecret: string;
+
+	constructor(
+		clientId: string,
+		clientSecret: string,
+		options?: {
+			redirectURI?: string;
+		}
+	) {
+		const baseUrl = "https://api.intra.42.fr";
+
+		const authorizeEndpoint = baseUrl + "/oauth/authorize";
```

Comment:
Super minor nit-pick but can you just do

```suggestion
		const authorizeEndpoint = "https://api.intra.42.fr/oauth/authorize";
```

and make it a global variable (see providers like Apple)